### PR TITLE
Fix a logical error on mutation

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Fixes `Part {} with metadata version {} contains column {} that is absent in table {} with metadata version {}` for old parts

Related to https://github.com/ClickHouse/ClickHouse/pull/49889 and https://github.com/ClickHouse/ClickHouse/pull/46946

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit f5327e79bb37c3cc061b8704ffdb85aa4f0b31c4 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51502/f5327e79bb37c3cc061b8704ffdb85aa4f0b31c4/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by yakov-olkhovskiy — APPROVED** [review]

**Comment by tavplubix** [comment]
Sorry, I had to update this PR because I've found another issue: the `metadata_version.txt` file was missing when cloning a part in mutations.

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -7286,19 +7301,22 @@ void StorageReplicatedMergeTree::replacePartitionFrom(
 
             UInt64 index = lock->getNumber();
             MergeTreePartInfo dst_part_info(partition_id, index, index, src_part->info.level);
-            MergeTreeData::HardlinkedFiles hardlinked_files;
-
-            bool copy_instead_of_hardlink = storage_settings_ptr->allow_remote_fs_zero_copy_replication
-                                            && src_part->isStoredOnRemoteDiskWithZeroCopySupport();
+            HardlinkedFiles hardlinked_files;
```
Comment (yakov-olkhovskiy) [inline src/Storages/StorageReplicatedMergeTree.cpp:7304]: I don't see `hardlinked_files` is used anywhere...

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 2]
```diff
@@ -7524,20 +7541,23 @@ void StorageReplicatedMergeTree::movePartitionToTable(const StoragePtr & dest_ta
             UInt64 index = lock->getNumber();
             MergeTreePartInfo dst_part_info(partition_id, index, index, src_part->info.level);
 
-            MergeTreeData::HardlinkedFiles hardlinked_files;
+            HardlinkedFiles hardlinked_files;
```
Comment (yakov-olkhovskiy) [inline src/Storages/StorageReplicatedMergeTree.cpp:7544]: I don't see `hardlinked_files` is used anywhere...

**Review by yakov-olkhovskiy — APPROVED** [review]
seems there are couple unused variables

**Pull request merged**